src/coreComponents/common/format/StringUtilities.hpp

```diff
@@ -66,14 +66,57 @@ string join( IT first, IT last, S const & delim = S() )
  * @brief Join strings or other printable objects with a delimiter.
  * @tparam CONTAINER type of container to join
  * @tparam S the type of delimiter, usually char, char const * or string
- * @param cont the container to join
+ * @param container the container to join
  * @param delim delimiter used to glue together strings
  * @return a string containing input values concatenated with a delimiter
  */
 template< typename CONTAINER, typename S = char >
-string join( CONTAINER const & cont, S const & delim = S() )
+string join( CONTAINER const & container, S const & delim = S() )
+{
+  return join( std::begin( container ), std::end( container ), delim );
+}
+
+/**
+ * @brief Join strings or other printable objects returned by a formatter functor.
+ * @tparam IT                type of iterator into the range of objects to join
+ * @tparam S                 type of delimiter, usually char, char const * or string
+ * @tparam LAMBDA            type of formatter functor, usually `[]( auto it ) -> string`
+ * @param iteratorFormatter  formatter function to get a formattable value from a IT iterator
+ * @param first              iterator to start of the range
+ * @param last               iterator past-the-end of the range
+ * @param delim              delimiter used to glue together strings
+ * @return a string containing input values concatenated with a delimiter
+ */
+template< typename IT, typename S, typename LAMBDA >
+string joinLamda( IT first, IT last, S const & delim, LAMBDA iteratorFormatter )
+{
+  if( first == last )
+  {
+    return {};
+  }
+  std::ostringstream oss;
+  oss << iteratorFormatter( first );
+  while( ++first != last )
+  {
+    oss << delim << iteratorFormatter( first );
+  }
+  return oss.str();
+}
+
+/**
+ * @brief Join strings or other printable objects returned by a formatter functor.
+ * @tparam CONTAINER         type of container to join
+ * @tparam S                 type of delimiter, usually char, char const * or string
+ * @tparam LAMBDA            type of formatter functor, usually `[]( auto it ) -> string`
+ * @param iteratorFormatter  formatter function to get a formattable value from an iterator of the container
+ * @param container          container to join
+ * @param delim              delimiter used to glue together strings
+ * @return a string containing input values concatenated with a delimiter
+ */
+template< typename CONTAINER, typename S, typename LAMBDA >
+string joinLamda( CONTAINER const & container, S const & delim, LAMBDA iteratorFormatter )
```

Comment:
_joinLambda_
_funcFormatter_ instead of _iteratorFormatter_  because the parameter is a function not an iterator